searx/infopage/__init__.py

```diff
@@ -171,9 +171,13 @@ def get_page(self, pagename: str, locale: typing.Optional[str] = None):
         self.CACHE[cache_key] = page
         return page
 
-    def all_pages(self, locale: typing.Optional[str] = None):
+    def iter_pages(self, locale: typing.Optional[str] = None, fallback_to_default=False):
         """Iterate over all pages of the TOC"""
         locale = locale or self.locale_default
-        for pagename in self.toc:
-            page = self.get_page(pagename, locale)
-            yield pagename, page
+        for page_name in self.toc:
+            page_locale = locale
+            page = self.get_page(page_name, locale)
+            if fallback_to_default and page is None:
+                page_locale = self.locale_default
+                page = self.get_page(page_name, self.locale_default)
+            yield page_name, page_locale, page
```

Comment:
Do note there is `page_locale` in addition to the existing `page_name` and `page`.